ledger/consumer/smartcontract/mod.go

```diff
@@ -0,0 +1,147 @@
+package smartcontract
+
+import (
+	proto "github.com/golang/protobuf/proto"
+	"go.dedis.ch/fabric/encoding"
+	"go.dedis.ch/fabric/ledger/consumer"
+	"go.dedis.ch/fabric/ledger/inventory"
+	"golang.org/x/xerrors"
+)
+
+//go:generate protoc -I ./ --go_out=./ ./messages.proto
+
+// Contract is an interface that provides the primitives to execute a smart
+// contract transaction and produce the resulting instance.
+type Contract interface {
+	// Spawn is called to create a new instance.
+	Spawn(ctx SpawnContext) (proto.Message, error)
+
+	// Invoke is called to update an existing instance.
+	Invoke(ctx InvokeContext) (proto.Message, error)
```

Comment:
no Delete?

Reply:
As per comment above, I didn't think about a logic when deleting but it might be yes. We can see later if it's needed.